src/main/java/net/sf/jabref/logic/integrity/IntegrityCheck.java

```diff
@@ -320,4 +326,32 @@ private BracketChecker(String field) {
         }
     }
 
+    private static class HashChecker implements Checker {
+
+        // Detect # if it doesn't have a \ in front of it or if it starts the string
+        private static final Pattern UNESCAPED_HASH = Pattern.compile("(?<!\\\\)#|^#");
+
+
+        /**
+         * Checks, if there is an even number of unescaped #
+         */
+        @Override
+        public List<IntegrityMessage> check(BibEntry entry) {
+            List<IntegrityMessage> results = new ArrayList<>();
+            for (String fieldName : entry.getFieldNames()) {
```

Comment:
I tried to find it, but either field names or field content as far as I can
tell. Would clearly be more convenient/cleaner.